src/utils/request.js

```diff
@@ -160,7 +160,7 @@ export const register = async (data) => {
   return json;
 };
 
-export const assignRole = async (data) => {
+export const assignCandidate = async (data) => {
```

Comment:
Instead of of splitting this into 2 separate functions, in the `Register.jsx`, you could send the role in the assignRole() function, and then in the `request.js` file you could use the conditional that we are using in other functions. This way we could keep our assignRole function and just tweak it a bit.

`const entity = role === "candidate" ? "candidates" : "companies"`